### PR TITLE
Implemented equals and hashcode for SqlParameter and JsonSerialzable

### PR DESCRIPTION
**Pull request by kushagraThapar**
* Implemented equals and hashcode for SqlParameter and JsonSerialzable
* Closes : https://github.com/Azure/azure-sdk-for-java/issues/14862

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/SqlParameterTest.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class SqlParameterTest {
+
+    @Test(groups = {"unit"})
+    public void testSqlParameterEquals() {
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/SqlParameterTest.java:16]: please drop the "test" prefix to be consistent with other test naming.

File: sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/SqlParameterTest.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        ArrayList<SqlParameter> sqlParameters = Lists.newArrayList(new SqlParameter("@param1", 3), new SqlParameter(
+            "@param2", 4));
+
+        assertThat(sqlParameters.contains(new SqlParameter("@param1", 3))).isTrue();
+        assertThat(sqlParameters.contains(new SqlParameter("@param2", 4))).isTrue();
+    }
+
+    @Test(groups = {"unit"})
+    public void testSqlParameterHashcode() {
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/SqlParameterTest.java:25]: ditto

**Review by moderakh — APPROVED** [review]
minor comment. 

other than that LGTM

File: sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/SqlParameterTest.java (hunk 1 above)
Comment (kushagraThapar) [reply sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/SqlParameterTest.java]: Done.

File: sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/SqlParameterTest.java (hunk 2 above)
Comment (kushagraThapar) [reply sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/SqlParameterTest.java]: Done, thanks :)

**Pull request merged**